host/app/components/card-catalog-editor.gts

```diff
@@ -0,0 +1,133 @@
+import { RealmPaths } from '@cardstack/runtime-common/paths';
+import { Loader } from '@cardstack/runtime-common/loader';
+//@ts-ignore cached not available yet in definitely typed
+import { tracked, cached } from '@glimmer/tracking';
+//@ts-ignore glint does not think this is consumed-but it is consumed in the template
+import { hash } from '@ember/helper';
+
+interface Signature {
+  Args: {
+    ref: ExportedCardRef;
+  }
+}
+
+export default class CardCatalogEditor extends Component<Signature> {
+  <template>
+    <ul>
+      {{#each this.catalogEntry.instances as |entry|}}
+        <li>
+          <LinkTo @route="application" @query={{hash path=(this.modulePath (ensureJsonExtension entry.id))}}>
+            {{entry.id}}
+          </LinkTo>
+          <fieldset>
+            <legend>Catalog Entry Editor</legend>
+            <ImportModule @url={{entry.meta.adoptsFrom.module}}>
+              <:ready as |module|>
+                <CardEditor
+                  @card={{hash type="existing" url=entry.id json=(hash data=entry) format="edit"}}
+                  @module={{module}}
+                  @onSave={{this.onSave}}
+                />
+              </:ready>
+              <:error as |error|>
+                <h2>Encountered {{error.type}} error</h2>
+                <pre>{{error.message}}</pre>
+              </:error>
+            </ImportModule>
+          </fieldset>
+          {{!-- TODO: Catalog Entry Preview --}}
+        </li>
+      {{else}}
+        {{#if this.showEditor}}
+          <fieldset>
+            <legend>Publish New Card Type</legend>
+            <ImportModule @url={{this.catalogEntryRef.module}}>
+              <:ready as |module|>
+                <CardEditor
+                  @card={{hash type="new" realmURL=this.localRealm.url.href cardSource=this.catalogEntryRef initialAttributes=this.catalogEntryAttributes}}
+                  @module={{module}}
+                  @onSave={{this.onSave}}
+                  @onCancel={{this.onCancel}}
+                />
+              </:ready>
+              <:error as |error|>
+                <h2>Encountered {{error.type}} error</h2>
+                <pre>{{error.message}}</pre>
+              </:error>
+            </ImportModule>
+          </fieldset>
+        {{else}}
+          <button {{on "click" this.displayEditor}} type="button">
+            Publish Card Type
+          </button>
+        {{/if}}
+      {{/each}}
+    </ul>
+  </template>
+
+  @service declare localRealm: LocalRealm;
+  @service declare router: RouterService;
+  catalogEntryRef: ExportedCardRef = {
+    module: 'https://cardstack.com/base/catalog-entry',
+    name: 'CatalogEntry',
+  };
+  catalogEntryAttributes = {
+    title: this.args.ref.name,
+    description: `Catalog entry for ${this.args.ref.name} type`,
+    ref: this.args.ref,
+  }
+  catalogEntry = getSearchResults(this, () => ({
+    filter: {
+      on: this.catalogEntryRef,
+      eq: { ref: this.args.ref },
+    },
+  }));
```

Comment:
it would be strange to have multiple catalog entries for a ref--probably we should just pick the first one that comes back